### DISCUSSION TITLE
Migrating to GLSP 2.0

### DISCUSSION THREAD

**Discussion by quratulain-ali** [eclipse-glsp/glsp#1210] [General]
I have been putting some effort in migrating my project to use GLSP 2.0.0. According to [this](https://github.com/eclipse-glsp/glsp-theia-integration/commit/591932b443533a799165ba8ac901a734fe67e2c2), the concept of
the `TheiaGLSPConnector` is removed. I have removed the connection classes and I am getting the following error.

> Failed to start the backend application:
> TypeError: Class extends value undefined is not a constructor or null
>     at Object.<anonymous> (/workspaces/electronics-modelserver-glsp/client/node_modules/@eclipse-glsp/theia-integration/lib/common/channel-connection.js:25:53)
>     at Module._compile (node:internal/modules/cjs/loader:1198:14)
>     at Object.Module._extensions..js (node:internal/modules/cjs/loader:1252:10)
>     at Module.load (node:internal/modules/cjs/loader:1076:32)
>     at Function.Module._load (node:internal/modules/cjs/loader:911:12)
>     at Module.require (node:internal/modules/cjs/loader:1100:19)
>     at require (node:internal/modules/cjs/helpers:119:18)
>     at Object.<anonymous> (/workspaces/electronics-modelserver-glsp/client/node_modules/@eclipse-glsp/theia-integration/lib/common/index.js:32:14)
>     at Module._compile (node:internal/modules/cjs/loader:1198:14)
>     at Object.Module._extensions..js (node:internal/modules/cjs/loader:1252:10)
> /workspaces/electronics-modelserver-glsp/client/node_modules/@theia/core/lib/node/main.js:22
>     throw reason;

What does this error mean

**Comment by tortmayr** [comment]
Hi @quratulain-ali,

unfortunately I'm not familiar with that Error so it's hard to give you any tips/input on that.
FYI the first iteration of our [Migration Guide](https://github.com/eclipse-glsp/glsp/blob/master/MIGRATION_GUIDE.md)  is now available it covers the migration steps for the client and Theia integration.
So maybe you could check it out to see if their are any migration steps that you missed.

Could you provide additional information regarding your setup?
- What node version are you using?
- What OS are you using?
- What Theia version are you using?
- Are you using a Java or Node server?
- Are you using additional components on top of GLSP (e.g. EMF.cloud ModelServer)?
